portality/scripts/link_checker_report.py

```diff
@@ -38,26 +46,84 @@ def write_results(df, filename='multi_result.csv'):
     print("Result CSV file has been written.")
 
 
-def fetch_matching_rows(df, report_values):
+def _get_link_type(link, journal):
+    form = JournalFormXWalk.obj2form(journal)
+
+    locations = []
+    subs = []
+    for k, v in form.items():
+        if v is None:
+            continue
+        if isinstance(v, list):
+            if link in v:
+                locations.append(k)
+            else:
+                for e in v:
+                    if isinstance(e, dict):
+                        for sk, sv in e.items():
+                            if not isinstance(sv, str):
+                                continue
+                            if link == sv:
+                                locations.append(sk)
+                            elif sv.startswith(link):
+                                subs.append(sk)
+                    else:
+                        if e.startswith(link):
+                            subs.append(k)
+                            break
+        else:
+            if not isinstance(v, str):
+                continue
+            if v == link:
+                locations.append(k)
+            elif v.startswith(link):
+                subs.append(k)
+
+    return locations + subs
+
+
+def fetch_matching_rows(journal_url_index, report_values):
     """Check with journals dataframe and retrieve matching rows with url.
        :param df: DataFrame
        :param report_values: url to match
        :return: DataFrame with matching rows
     """
     # Search for the text in the entire csv file
-    mask = df.applymap(lambda x: report_values["url"] in str(x))
+    #mask = df.applymap(lambda x: report_values["url"] in str(x))
 
     # Get the rows where the text is found
-    df_result = df[mask.any(axis=1)]
+    #df_result = df[mask.any(axis=1)]
+    journal_data = journal_url_index.get(report_values["url"])
 
-    if not df_result.empty:
-        columns = ['Journal title', 'Added on Date', 'Last updated Date']
+    # if not df_result.empty:
+    if journal_data is not None:
+        # columns = ['Journal title', 'Added on Date', 'Last updated Date', "Journal ID"]
 
         # Select the desired columns from the DataFrame
-        df_result_selected_columns = df_result[columns].copy()  # create a copy to avoid SettingWithCopyWarning
+        # df_result_selected_columns = df_result[columns].copy()  # create a copy to avoid SettingWithCopyWarning
+        df_result_selected_columns = pd.DataFrame(
+            data=[list(journal_data)],
+            columns=['Journal title', 'Added on Date', 'Last updated Date', "Journal ID"]
+        )
+
+        jid = df_result_selected_columns["Journal ID"].values[0]
+        journal = models.Journal.pull(jid)
```

Comment:
A pull from the index. Isn't this based on some version of the journalCSV, so this shouldn't be necessary - data should already be present / augment if necessary rather than going back to the index.

Reply:
We need the journal object itself to xwalk to the form model, to create the connection back to the form.  We can't construct a journal object from the CSV data